src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -967,7 +967,7 @@ private EnvVars getEnvVars(DescribableList<NodeProperty<?>, NodePropertyDescript
      */
     private void verifyNoHeaderJunk(TaskListener listener) throws IOException, InterruptedException {
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        connection.exec("echo hello > null",baos);
+        connection.exec("exit 0",baos);
```

Comment:
I did not try the most obvious choice...
This also works for Windows and Linux agents at the same time and does not leave garbage files on the remote host.